ios-base/View/Onboarding/SignUpViewController.swift

```diff
@@ -50,14 +49,37 @@ class SignUpViewController: UIViewController {
   }
   
   @IBAction func tapOnSignUpButton(_ sender: Any) {
-    UIApplication.showNetworkActivity()
-    viewModel.signup(success: { [unowned self] in
+    viewModel.signup()
+  }
+  
+  func setSignedUpRoot() {
+    let homeVC = self.storyboard?.instantiateViewController(withIdentifier: "HomeViewController") as? HomeViewController
+    UIApplication.shared.keyWindow?.rootViewController = homeVC
+  }
+  
+  func enableSignUpButton(_ enable: Bool) {
+    signUp.alpha = enable ? 1 : 0.5
+    signUp.isEnabled = enable
+  }
+}
+
+extension SignUpViewController: SignUpViewModelDelegate {
+  func formDidChange() {
+    enableSignUpButton(viewModel.hasValidData)
```

Comment:
Why not just calling `enableSignUpButton ()` and put the `viewModel.hasValidData` inside that method?

Reply:
Because I want to be able to force a value in some cases. i.e: line 27

Reply:
Every time you force that value is with `false`? Maybe a default `false` there would be nice. (My OCD says that `..(false/true)` is bad for my mental health haha) 🙈

Reply:
implementing agreed solution on #96